.github/workflows/codeql-analysis.yml

```diff
@@ -39,11 +39,11 @@ jobs:
 
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
       # Initializes the CodeQL tools for scanning.
       - name: Initialize CodeQL
-        uses: github/codeql-action/init@v1
+        uses: github/codeql-action/init@v3
         with:
           languages: ${{ matrix.language }}
```

Comment:
You could inline this Matrix in the same way you did for the Node Versions, just to be consistent

Reply:
Good spot - have done this